crates/oxc_linter/src/rules/jsx_a11y/aria_activedescendant_has_tabindex.rs

```diff
@@ -0,0 +1,147 @@
+use oxc_ast::{
+    ast::{JSXAttribute, JSXAttributeItem},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{
+    context::LintContext,
+    globals::HTML_TAG,
+    rule::Rule,
+    utils::{get_element_type, has_jsx_prop_lowercase, is_interactive_element, parse_jsx_value},
+    AstNode,
+};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-jsx-a11y(aria-activedescendant-has-tabindex): Enforce elements with aria-activedescendant are tabbable.")]
+#[diagnostic(
+    severity(warning),
+    help("An element that manages focus with `aria-activedescendant` must have a tabindex.")
+)]
+struct AriaActivedescendantHasTabindexDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct AriaActivedescendantHasTabindex;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Enforce elements with aria-activedescendant are tabbable.
+    ///
+    /// ### Example
+    /// ```jsx
+    /// // Good
+    /// <CustomComponent />
+    /// <CustomComponent aria-activedescendant={someID} />
+    /// <CustomComponent aria-activedescendant={someID} tabIndex={0} />
+    /// <CustomComponent aria-activedescendant={someID} tabIndex={-1} />
+    /// <div />
+    /// <input />
+    /// <div tabIndex={0} />
+    /// <div aria-activedescendant={someID} tabIndex={0} />
+    /// <div aria-activedescendant={someID} tabIndex="0" />
+    /// <div aria-activedescendant={someID} tabIndex={1} />
+    /// <div aria-activedescendant={someID} tabIndex={-1} />
+    /// <div aria-activedescendant={someID} tabIndex="-1" />
+    /// <input aria-activedescendant={someID} />
+    /// <input aria-activedescendant={someID} tabIndex={0} />
+    /// <input aria-activedescendant={someID} tabIndex={-1} />
+    ///
+    /// // Bad
+    /// <div aria-activedescendant={someID} />
+    /// ```
+    AriaActivedescendantHasTabindex,
+    correctness
+);
+
+impl Rule for AriaActivedescendantHasTabindex {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::JSXOpeningElement(jsx_opening_el) = node.kind() else {
+            return;
+        };
+
+        if has_jsx_prop_lowercase(jsx_opening_el, "aria-activedescendant").is_none() {
+            return;
+        };
```

Comment:
i think we can remove this check?

We check again below.

Reply:
I'm sorry, I couldn't figure out where the duplication is on my own. Which line is conducting the same check?

Reply:
l 81

` has_jsx_prop_lowercase(jsx_opening_el, "tabIndex")`

Reply:
wait ignore me

Reply:
lol